Dockerfile.planner

```diff
@@ -0,0 +1,8 @@
+#planner stage
```

Comment:
Should we keep image with cargo cached dependencies separately from Linux Arch based `ci-image`?

Reply:
Unfortunately, this takes us back to the architecture that we moved away from. I don't think that doing things this way is tenable. It's preferable for us to compile the binary separately and simply copy it into the container and produce an image.

Reply:
Do you mean to combine Arch based ci-image and `cargo chef` in one Dockerfile.base file? Or to join `cargo chef` and `cargo build` together? In this case, we have to pay attention to `Cargp.lock` change trigger and where to use it properly.

Reply:
> Do you mean to combine Arch based ci-image and cargo chef in one Dockerfile.base file?

Yes. 

> In this case, we have to pay attention to Cargo.lock change trigger and where to use it properly.

We should just have a manual workflow trigger.  This is  needed anyway so that we can update the image when there's a new toolchain. 

Cargo lock changes far more often than is necessary to re-run this operation. If one line changed, it makes the compilation process take 1/600th longer, if that. Re-caching the dependencies only makes sense if we have a large rewrite and version bump all of the external dependencies. Plus if `Cargo.lock` is slightly outdated, the compilation will take only slightly longer.

Reply:
> > Do you mean to combine Arch based ci-image and cargo chef in one Dockerfile.base file?
> 
> Yes.
> 
It was combined in one `Dockerfile.base`